clang/lib/DPCT/AnalysisInfo.cpp

```diff
@@ -4280,7 +4289,13 @@ std::string getStringForRegexDefaultQueueAndDevice(HelperFuncType HFT,
         DpctGlobalInfo::getTempVariableDeclCounterMap().find(CounterKey);
     if (TempVariableDeclCounterIter ==
         DpctGlobalInfo::getTempVariableDeclCounterMap().end()) {
-      return getDefaultString(HFT);
+      return getDefaultString(HFT, FilePath);
+    }
+
+    if (HFT == HelperFuncType::HFT_DefaultQueue) {
+      requestFeature(HelperFeatureEnum::Device_get_default_queue, FilePath);
```

Comment:
The requestFeature should not be here.

Reply:
Should modify SYCLomatic/clang/lib/DPCT/AnalysisInfo.cpp line:352:
```c++
  if (!getDeviceChangedFlag() && getUsingDRYPattern()) {
    bool NeedDpctHelpFunc = DpctGlobalInfo::needDpctDeviceExt() ||
                            TempVariableDeclCounterMap.size() > 1 ||
                            DpctGlobalInfo::getUsmLevel() == UsmLevel::UL_None;
    HelperFeatureEnum DeviceFeatureEnum =
        HelperFeatureEnum::Device_get_current_device;
    HelperFeatureEnum QueueFeatureEnum =
        HelperFeatureEnum::Device_get_default_queue;

    unsigned int IndentLen = 2;
    if (getGuessIndentWidthMatcherFlag())
      IndentLen = getIndentWidth();
    std::string IndentStr = std::string(IndentLen, ' ');
    std::string DevDeclStr = getNL() + IndentStr;
    llvm::raw_string_ostream DevDecl(DevDeclStr);
    std::string QDeclStr =
        getNL() + IndentStr + MapNames::getClNamespace() + "queue ";
    llvm::raw_string_ostream QDecl(QDeclStr);
    if (NeedDpctHelpFunc) {
      DevDecl << MapNames::getDpctNamespace()
              << "device_ext &dev_ct1 = " << MapNames::getDpctNamespace()
              << "get_current_device();";
      QDecl << "&q_ct1 = dev_ct1.default_queue();";
    } else {
      DevDecl << MapNames::getClNamespace() + "device dev_ct1;";
      // Now the UsmLevel must not be UL_None here.
      QDecl << "q_ct1(dev_ct1, " << MapNames::getClNamespace()
            << "property_list{" << MapNames::getClNamespace()
            << "property::queue::in_order()";
      if (DpctGlobalInfo::getEnablepProfilingFlag()) {
        QDecl << ", " << MapNames::getClNamespace()
              << "property::queue::enable_profiling()";
      }
      QDecl << "});";
    }

    for (auto &Counter : TempVariableDeclCounterMap) {
      const auto ColonPos = Counter.first.find_last_of(':');
      const auto DeclLocFile = Counter.first.substr(0, ColonPos);
      const auto DeclLocOffset = std::stoi(Counter.first.substr(ColonPos + 1));
      if (Counter.second.CurrentDeviceCounter > 1 ||
          Counter.second.DefaultQueueCounter > 1) {
        Counter.second.PlaceholderStr[2] = "dev_ct1";
        getInstance().addReplacement(std::make_shared<ExtReplacement>(
            DeclLocFile, DeclLocOffset, 0, DevDecl.str(), nullptr));
        if (!NeedDpctHelpFunc) {
          DeviceFeatureEnum = HelperFeatureEnum::no_feature_helper;
        }
        if (Counter.second.DefaultQueueCounter > 1 || !NeedDpctHelpFunc) {
          Counter.second.PlaceholderStr[1] = "q_ct1";
          getInstance().addReplacement(std::make_shared<ExtReplacement>(
              DeclLocFile, DeclLocOffset, 0, QDecl.str(), nullptr));
          if (!NeedDpctHelpFunc) {
            DeviceFeatureEnum = HelperFeatureEnum::no_feature_helper;
          }
        }
      }
      if (Counter.second.CurrentDeviceCounter > 0 ||
          Counter.second.DefaultQueueCounter > 1)
        requestFeature(DeviceFeatureEnum, DeclLocFile);
      if (Counter.second.DefaultQueueCounter > 0)
        requestFeature(QueueFeatureEnum, DeclLocFile);
    }
  }
```

Reply:
updated.